frontend/src/metabase/visualizations/types/visualization.ts

```diff
@@ -150,6 +149,12 @@ export type Visualization = React.ComponentType<VisualizationProps> & {
     settings: VisualizationSettings,
     query: Query,
   ) => void | never;
-  isLiveResizable: (series: Series) => boolean;
+  isLiveResizable?: (series: Series) => boolean;
   onDisplayUpdate?: (settings: VisualizationSettings) => VisualizationSettings;
 };
+
+// TODO: add component property for the react component instead of the intersection
```

Comment:
Do you plan on doing this in this PR or later?

Reply:
This is sasha’s comment that I just moved down.  I’m not sure what it means.

Reply:
Ah got it. It looks like you ended up doing this though since you made the `Visualization` type based on `React.ComponentType<VisualizationProps>`. I think we can remove this comment now, cc @alxnddr for confirmation.

Reply:
@shaunlebron I wonder why you did not add `name: string;` property to the `Visualization` type which looks like the shortest path, wdyt about it? The `VisualizationProperties` looks a bit ambiguous in relation to VisualizationProps.

The idea behind my TODO comment was to separate visualization definitions from the component: replace `Object.assign(component, { checkRenderable: () => {}, ... })` with `{ component: PieChart, checkRenderable: () => {}, ... }`